packages/experimental/Badge/src/colorHelper.ts

```diff
@@ -0,0 +1,209 @@
+import { ColorValue } from 'react-native';
+import { Theme } from '@fluentui-react-native/framework';
+
+export type BadgeColors = {
+  backgroundColor?: ColorValue;
+  color?: ColorValue;
+  iconColor?: ColorValue;
+  borderColor?: ColorValue;
+  backgroundColorDark?: ColorValue;
+  colorDark?: ColorValue;
+  borderColorDark?: ColorValue;
+  hcBackground?: ColorValue;
+  hcColor?: ColorValue;
+  hcBorderColor?: ColorValue;
+};
+
+type ColorProps = {
+  backgroundColor?: ColorValue;
+  color?: ColorValue;
+  iconColor?: ColorValue;
+  borderColor?: ColorValue;
+  hcBackground?: ColorValue;
+  hcColor?: ColorValue;
+  hcBorderColor?: ColorValue;
+};
+
+type ThemeProps = {
+  light?: ColorProps;
+  dark?: ColorProps;
+  hc?: ColorProps;
+};
+
+export function getHCProps(theme: Theme, colors?: BadgeColors) {
+  const hcColors = colors || {};
+  const hcBackground = hcColors.hcBackground || theme.colors.transparentBackground;
+  const hcColor = hcColors.hcColor || theme.colors.neutralForeground3;
+  const hcBorderColor = hcColors.hcBorderColor || theme.colors.neutralForeground3;
+
```

Comment:
nit: to me it feels out of place for hc to have its own props function, when this information could be included in each badge appearance's function instead. Then all of a particular badge appearance's color information is in one place, instead of spread out over two, and hc works like the other modes/themes.

Reply:
yeah, I'll move it to appearance function. This HC function was created before I moved everything to one function